server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -896,6 +903,15 @@ public void relocated(
             // Fail primary relocation source and target shards.
             failShard("timed out waiting for relocation hand-off to complete", null);
             throw new IndexShardClosedException(shardId(), "timed out waiting for relocation hand-off to complete");
+        } finally {
+            // If the primary mode is still true after the end of handoff attempt, it basically means that the relocation
+            // failed. The existing primary will continue to be the primary, so we need to allow the segments and translog
+            // upload to resume.
+            if (replicationTracker.isPrimaryMode()) {
+                for (Releasable releasable : releasablesOnNoHandoff) {
+                    releasable.close();
+                }
```

Comment:
You can use the utilitity `Releasables.close`

Reply:
Should we move this `release` part to `abortRelocationHandoff` ? This is a much better fit there .

Reply:
The abortRelocationHandoff might not run if the failure happens before the control reaches the below code -
https://github.com/opensearch-project/OpenSearch/blob/5bb6caec906f9e89d330332ebb74789571409eb1/server/src/main/java/org/opensearch/index/shard/IndexShard.java#L877-L890

The failures can happen during draining refreshes or translog sync. With this being in finally, we ensure that the rollback step runs irrespective of where the failure occurred.